### PR TITLE
Munge externals

### PR DESCRIPTION
**Pull request by jassigill2000**
Changes made to make mungefs into an external.

### CODE REVIEW

File: irods_consortium_continuous_integration_build_hook.py [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+    #irods_python_ci_utilities.install_os_packages(get_build_prerequisites())
+
+def install_build_prerequisites_yum():
+    if irods_python_ci_utilities.get_distribution() == 'Centos' or irods_python_ci_utilities.get_distribution() == 'Centos linux':
+        irods_python_ci_utilities.subprocess_get_output(['sudo', 'systemctl', 'restart', 'autofs'], check_rc=True)
+        if irods_python_ci_utilities.get_distribution_version_major() == '6':
+            os.environ['CC'] = '/opt/rh/devtoolset-6/root/usr/bin/gcc'
+            os.environ['CXX'] = '/opt/rh/devtoolset-6/root/usr/bin/g++'
+            os.environ['PATH'] = '/opt/rh/devtoolset-6/root/usr/bin' + os.pathsep + os.environ['PATH']
+    #irods_python_ci_utilities.install_os_packages(get_build_prerequisites())
```
Comment (alanking) [inline irods_consortium_continuous_integration_build_hook.py]: Given that we will not be supporting Cen6 or Ub12, those sections can be removed. This effectively eliminates install_build_prerequisites_* except for lines 41 and 42 which is where the action happens. Can we move these two lines to main() or to a well-named function and delete the rest?

File: irods_consortium_continuous_integration_build_hook.py [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+import multiprocessing
+import optparse
+import os
+import tempfile
+
+import irods_python_ci_utilities
+
+
+def get_build_prerequisites_all():
+    pass;
```
Comment (trel) [inline irods_consortium_continuous_integration_build_hook.py]: semicolon

File: irods_consortium_continuous_integration_build_hook.py [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+
+def get_build_prerequisites_apt():
+    pass;
```
Comment (trel) [inline irods_consortium_continuous_integration_build_hook.py]: semicolon

File: irods_consortium_continuous_integration_build_hook.py [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+
+def get_build_prerequisites_yum():
+    pass;
```
Comment (trel) [inline irods_consortium_continuous_integration_build_hook.py]: semicolon

File: irods_consortium_continuous_integration_build_hook.py [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+    except KeyError:
+        irods_python_ci_utilities.raise_not_implemented_for_distribution()
+
+def install_build_prerequisites_apt():
+    if irods_python_ci_utilities.get_distribution() == 'Ubuntu': # cmake from externals requires newer libstdc++ on ub12
+        if irods_python_ci_utilities.get_distribution_version_major() == '12':
+            irods_python_ci_utilities.install_os_packages(['python-software-properties'])
+            irods_python_ci_utilities.subprocess_get_output(['sudo', 'add-apt-repository', '-y', 'ppa:ubuntu-toolchain-r/test'], check_rc=True)
+            irods_python_ci_utilities.install_os_packages(['libstdc++6'])
+    #irods_python_ci_utilities.install_os_packages(get_build_prerequisites())
```
Comment (trel) [inline irods_consortium_continuous_integration_build_hook.py]: remove commented code

File: irods_consortium_continuous_integration_build_hook.py [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+
+def install_build_prerequisites_yum():
+    if irods_python_ci_utilities.get_distribution() == 'Centos' or irods_python_ci_utilities.get_distribution() == 'Centos linux':
+        irods_python_ci_utilities.subprocess_get_output(['sudo', 'systemctl', 'restart', 'autofs'], check_rc=True)
+        if irods_python_ci_utilities.get_distribution_version_major() == '6':
```
Comment (trel) [inline irods_consortium_continuous_integration_build_hook.py]: no more centos6

File: install_prerequisites.py [hunk 7]
```diff
@@ -104,6 +106,8 @@ def main():
                'bzip2-devel','libcurl-devel','libxml2-devel','libtool','libuuid-devel','openssl-devel']
         build.run_cmd(cmd, check_rc='installing prerequisites failed')
         # get necessary ruby gems
+        cmd = ['sudo','gem','install','rake']
+        build.run_cmd(cmd, check_rc='installing rake failed')
```
Comment (alanking) [inline install_prerequisites.py]: Nitpick: Ordering on gem installations between apt and yum is inconsistent

File: versions.json [hunk 8]
```diff
@@ -250,6 +250,22 @@
         ],
         "fpm_directories": ["bin","include","lib"]
     },
+    "mungefs": {
+        "commitish": "2d7a4f6ff56e64b560f582a21438162a1e8f36ab",
```
Comment (trel) [inline versions.json]: we should have a release tag of mungefs to use from this particular commit (currently tip of master) - need to release that as v1.0.2 and reference that here instead...

**Comment by trel** [comment]
is this ready to be merged?   or we're still waiting for the new mungefs tag?

**Comment by trel** [comment]
looks good - add # to both commits

**Pull request merged**